pkg/services/ngalert/schedule/schedule.go

```diff
@@ -392,6 +397,11 @@ func (sch *schedule) schedulePeriodic(ctx context.Context) error {
 			// so, at the end, the remaining registered alert rules are the deleted ones
 			registeredDefinitions := sch.registry.keyMap()
 
+			// While these are the rules that we iterate over, at the moment there's no 100% guarantee that they'll be
```

Comment:
I thought this was the case too but I now think it's no longer true. Even if a rule is removed we might still schedule it because `sch.schedulableAlertRules.all()` returns a new slice of schedulable alert rules. This means that if the rule is deleted after `schedulePeriodic` has called `sch.schedulableAlertRules.all()` then the deleted rule will be evaluated for at least one tick.

Reply:
Could you please make a suggestion on how I should word this then?

Reply:
I think the comment can just be removed?